controllers/vspherepolicy/policyevaluation/policyevaluation_controller.go

```diff
@@ -540,6 +545,14 @@ func (r *Reconciler) addComputePolicyRef(
 		return fmt.Errorf("failed to get compute policy: %w", err)
 	}
 
+	matches, err := matchesPolicy(obj, pol)
+	if err != nil {
+		return err
+	}
+	if !matches {
+		return fmt.Errorf("compute policy %q does not match", pol.Name)
```

Comment:
I don't think this is an error. I think it is expected for some mandatory policies to not be applicable to every VM so I wouldn't think about it as an error and wouldn't log it as such. Maybe this function could return `(bool, error)` with the bool indicating if policy was applied or not? That'd allow to provide more detailed log message without defining a specialized error type (and a specialized error type would be my alternative suggestion - just to make sure the log captures that it's not really an error, just that a policy didn't match).

Reply:
So, this is called in the `addComputePolicyRef()` which is called from the `reconcileExplicitPolicies()`. For mandatory polices, we skip them if not matched ([ref](https://github.com/vmware-tanzu/vm-operator/pull/1284/files#diff-644b778227d7912d308b6cca5ed285885c204429ed06771f8c693f05052c8b69R213-R221)). As we discussed offline, we're returning an error for not matched policies only if they're explicitly defined in `PolicyEvaluation.Spec.Policies` (populated from `VM.Spec.Polices`).